src/overlays/actors/ovl_En_Rz/z_en_rz.c

```diff
@@ -42,80 +39,646 @@ const ActorInit En_Rz_InitVars = {
+    Math_Vec3f_Copy(&this->actor.world.pos, &oldPos);
+    this->actor.scale.x = 0.01f;
+}
+
+void EnRz_ChangeAnim(PlayState* play, EnRz* this, s16 animIndex, u8 animMode, f32 transitionRate) {
+    f32 endFrame;
+    AnimationHeader** animationPtr;
+
+    if (!EN_RZ_GET_PARAM_8000(&this->actor)) {
+        animationPtr = D_80BFCD20;
+    } else {
+        animationPtr = D_80BFCD3C;
+    }
+
+    if ((animIndex >= 0) && (animIndex < 9) && ((animIndex != this->animIndex) || (animMode != ANIMMODE_LOOP))) {
+        if (animIndex >= ARRAY_COUNT(D_80BFCD20)) {
+            endFrame = Animation_GetLastFrame(sLinkAnimations[animIndex - ARRAY_COUNT(D_80BFCD20)]);
+            if (animMode == ANIMMODE_LOOP) {
+                LinkAnimation_Change(play, &this->skelAnime, sLinkAnimations[animIndex - ARRAY_COUNT(D_80BFCD20)],
+                                     2.0f / 3.0f, 0.0f, endFrame, ANIMMODE_LOOP, transitionRate);
+            } else {
+                LinkAnimation_Change(play, &this->skelAnime, sLinkAnimations[animIndex - ARRAY_COUNT(D_80BFCD20)],
+                                     2.0f / 3.0f, 0.0f, endFrame, ANIMMODE_LOOP, transitionRate);
+            }
+        } else {
+            Animation_Change(&this->skelAnime, animationPtr[animIndex], 1.0f, 0.0f,
+                             Animation_GetLastFrame(animationPtr[animIndex]), animMode, transitionRate);
+        }
+
+        this->animIndex = animIndex;
+    }
+}
+
+void func_80BFB9E4(PlayState* play, EnRz* this, s16 animIndex) {
+    EnRz_ChangeAnim(play, this, animIndex, ANIMMODE_LOOP, -5.0f);
+}
+
+void func_80BFBA1C(PlayState* play, EnRz* this, s16 animIndex) {
+    EnRz_ChangeAnim(play, this, animIndex, ANIMMODE_LOOP, 0.0f);
+}
+
+s32 func_80BFBA50(EnRz* this, PlayState* play) {
+    if (EN_RZ_GET_PATH(&this->actor) != 0x3F) {
+        this->path = &play->setupPathList[EN_RZ_GET_PATH(&this->actor)];
+        if (this->path != NULL) {
+            Path* path = this->path;
+            Vec3s* points = (Vec3s*)Lib_SegmentedToVirtual(path->points);
+            f32 temp_fv0 = points->x - this->actor.world.pos.x;
+            f32 temp_fv1 = points->z - this->actor.world.pos.z;
+
+            if ((SQ(temp_fv0) + SQ(temp_fv1)) < SQ(10.0f)) {
+                this->unk_3BC = 0;
+                this->unk_420 &= ~1;
+            } else {
+                this->unk_3BC = path->count - 1;
+                this->unk_420 |= 1;
+                return true;
+            }
+        }
+    } else {
+        this->path = NULL;
+        this->unk_3BC = 0;
+    }
+    return false;
+}
+
+s32 func_80BFBB44(EnRz* this) {
+    Path* path = this->path;
+    Vec3s* pointPos;
+    f32 temp_fa0;
+    f32 temp_fa1;
+
+    if (path == NULL) {
+        return true;
+    }
+
+    pointPos = &((Vec3s*)Lib_SegmentedToVirtual(path->points))[this->unk_3BC];
+    temp_fa0 = pointPos->x - this->actor.world.pos.x;
+    temp_fa1 = pointPos->z - this->actor.world.pos.z;
```

Comment:
This pattern appears in other actors, e.g.
```c

    curPoint = (Vec3s*)Lib_SegmentedToVirtual(path->points);
    curPoint += this->curPointIndex;
    diffX = curPoint->x - this->picto.actor.world.pos.x;
    diffZ = curPoint->z - this->picto.actor.world.pos.z;
    this->picto.actor.world.rot.y = Math_Atan2S(diffX, diffZ);
    Math_SmoothStepToS(&this->picto.actor.shape.rot.y, this->picto.actor.world.rot.y, 2, 0x7D0, 0xC8);
```
in EnGe2, I'd prefer if you used the same names.